cache-core/src/main/java/io/micronaut/cache/interceptor/CacheInterceptor.java

```diff
@@ -133,7 +137,8 @@ public Object intercept(MethodInvocationContext<Object, Object> context) {
                     CacheOperation cacheOperation = getCacheOperation(context,
                                                                       returnType.isVoid() || returnTypeValue
                                                                               .equalsType(Argument.VOID_OBJECT));
-                    if (cacheOperation.cacheable) {
+                    boolean cacheableCondition = context.findAnnotation(Cacheable.class).flatMap(a -> a.booleanValue(MEMBER_CONDITION)).orElseGet(() -> true);
```

Comment:
I think we should add some log statements when there is a condition but evaluates to false.

Reply:
this should be gated behind `if (context.isPresent(Cacheable.class, "condition"))` then it should use `context.booleanValue(Cacheable, "condition").orElse(false)`

which will make it more performant

Reply:
@sdelamo done in https://github.com/micronaut-projects/micronaut-cache/pull/692/commits/4d031b5a76e5fbef151eae470874a29c0e2f03dc